cluster/src/main/scala/io/snappydata/gemxd/SparkSQLExecuteImpl.scala

```diff
@@ -352,7 +386,15 @@ object SparkSQLExecuteImpl {
           val sb = new StringBuilder()
           Utils.dataTypeStringBuilder(other, sb)(col)
           // write the full length as an integer
-          hdos.writeUTF(sb.toString(), true, false)
+          val utf8String = UTF8String.fromString(sb.toString())
```

Comment:
Ok. Got it. There are lots of them which will never be used and we just need it for identifying complex type.